### DISCUSSION TITLE
Threading/ProcessPooling Backtest.Optimize

### DISCUSSION THREAD

**Discussion by KitHaywood** [kernc/backtesting.py#620] [General]
Hi There, 

I am trying to perform an optimised backtest on my user defined strategy. The scope of potential combinations for these 2 indicators is large, and therefore the cost-to-time for these computations is high. 

Is there a way to thread/process-pool backtesting.optimize? I'm currently thinking that I can chunk the indicator combinations up and then backtest.optimize again on the set of outcomes of the chunks - do you think this is the best way forward?

Any advice on how to thread/processpool would be very helpful,

Thanks, 

Kit

**Comment by kernc** [comment]
> chunk the indicator combinations up

This is how `Backtest.optimize(method='grid')` works internally. Unfortunately not so well on Windos. Ideally, someone would have a look at implementing proper multiprocessing on Widows:
https://github.com/kernc/backtesting.py/blob/94d20da85e278102a0cc71b27a3a35b815e11648/backtesting/backtesting.py#L1268-L1269
https://github.com/kernc/backtesting.py/blob/94d20da85e278102a0cc71b27a3a35b815e11648/backtesting/backtesting.py#L1353-L1382

**Reply by KitHaywood** [reply]
I'll give that a go, I'm running it on linux so should work, thanks

**Comment by ramzerof** [comment]
Note for Mac users:
the default start method on Mac is spawn, so the multiprocessing will not work on Mac out of the box.
To enable multiprocessing on Mac set the start method manually to 'fork' before running bt

```
import multiprocessing

multiprocessing.set_start_method("fork")
```
